agents/mysql/perfschema/perfschema.go

```diff
@@ -206,8 +220,57 @@ func (m *PerfSchema) runHistoryCacheRefresher(ctx context.Context) {
 	}
 }
 
+func getMySQLVersion(q *reform.Querier) (mysqlVersion float64, vendor string, err error) {
+	var name, ver string
+	err = q.QueryRow(fmt.Sprintf(`SHOW /* %s */ GLOBAL VARIABLES WHERE Variable_name = 'version'`, queryTag)).Scan(&name, &ver)
+	if err != nil {
+		return
+	}
+
+	var ven string
+	err = q.QueryRow(fmt.Sprintf(`SHOW /* %s */ GLOBAL VARIABLES WHERE Variable_name = 'version_comment'`, queryTag)).Scan(&name, &ven)
+	if err != nil {
+		return
+	}
+
+	ver, vendor = version.ParseMySQLVersion(ver, ven)
+	mysqlVersion, err = strconv.ParseFloat(ver, 64)
+	return
+}
+
 func (m *PerfSchema) refreshHistoryCache() error {
-	current, err := getHistory(m.q)
+	var mysqlVersion float64
+	var vendor string
+	mutexRW.RLock()
+	if v, ok := versions[m.agentID]; ok {
```

Comment:
no need to keep it in `versions` map, you could create new field in `PerfSchema` which would contain version and vendor of DB Server

Reply:
Fixed.